### PR TITLE
274 csv parser

### LINKED ISSUE #274: create csv parser script
- [x] take csv from discord channel (commit a dummy template only)
- [x] pass it into an internal representation and console.log
- [ ] validation of correct input

### PR DESCRIPTION
**Pull request by ardo715**
**Issue**

Need to be able to parse through X amount of students from csv into the front end

Closes #274 

**Solution**

Made a parseCSV function with a test script. The function also does integrity checks on the data being parsed.

**Discussion**
Merged master into the branch so should not be many conflicts

Currently the function will error if there is an invalid student in the csv file. This should be updated to create all the other students, and return a log of students that were unable to be created.

Currently uses a student class defined in the utils. This should be changed in the future to use the compatible types

**Risk**

No risk, all files are newly created.
Might have a problem with yarn / npm update as this uses the module csv-parse (npm install csv-parse)

**Reviewers**
@johnchen383 
@luorixo

### CODE REVIEW

**Review by bcho892 — COMMENTED** [review]
Looks good in terms of functionality and catches errors properly, however I think this could be done using typescript types instead of classes so the code is shorter:

```
type Student = {
  firstName: string;
  middleName: string;
  surname: string;
  email: string;
  yearLevel: number;
  heardFrom: string;
  reasonForTaking: string;
  teacherCode: string;
  teacherName: string;
  teacherEmail: string;
  schoolName: string;
  schoolAddress: string;
  phoneNumber: string;
  teacherCategory: string;
  island: Island;
  city: string;
};
``` 
and then it would be the same after, but you just say that:

```
const student: Student = {
  firstName: record.StudentFirstName,
  middleName: record.StudentMiddleName,
  surname: record.StudentSurname,
  email: record.StudentEmail,
  yearLevel: parseInt(record.StudentYearLevel),
  heardFrom: record.HeardFrom,
  reasonForTaking: record.ReasonForTaking,
  teacherCode: record.TeacherCode,
  teacherName: record.TeacherName,
  teacherEmail: record.TeacherEmail,
  schoolName: record.SchoolName,
  schoolAddress: record.SchoolAddress,
  phoneNumber: record.PhoneNumber,
  teacherCategory: record.TeacherCategory,
  island: record.Island as Island,
  city: record.City,
};
```
then like before:

```
students.push(student)
```
Also note that you won't need to call 
```
students.getAttributes
```
anymore and can just do 

```
console.log(students)
```

**Review by bcho892 — APPROVED** [review]
LGTM good job

File: client/src/utils/csv_parser.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import {parse} from '../../../node_modules/csv-parse';
```
Comment (johnchen383) [inline client/src/utils/csv_parser.ts:3]: you shouldn't have to reference a package like this. Does `import {parse} from 'csv-parse'` not work?

File: client/src/utils/csv_parser.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+
+// Island enums for error checking
+enum Island {
+    NORTH_ISLAND = 'North Island',
+    SOUTH_ISLAND = 'South Island'
+  }
+
+// Student object that can be accessed
```
Comment (johnchen383) [inline client/src/utils/csv_parser.ts:12]: Could you add a `TODO:` comment to note that this type will be eventually be replaced with a DTO class as defined in `common`

File: client/src/utils/csv_parser.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+function isValidYearLevel(yearLevel: number): boolean {
+  return yearLevel >= 1 && yearLevel <= 13;
+}
+
+function isValidEmail(email: string): boolean {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+}
```
Comment (johnchen383) [inline client/src/utils/csv_parser.ts:144]: awesome stuff

**Review by johnchen383 — CHANGES_REQUESTED** [review]
Great work. Some minor changes requested. I also not that there's no package.json file committed. Where is that change?

File: package.json [hunk 4]
```diff
@@ -9,5 +9,9 @@
         "service": "cd service && yarn run dev",
         "client": "cd client && yarn run dev"
     },
-    "packageManager": "yarn@1.22.19"
+    "packageManager": "yarn@1.22.19",
+    "dependencies": {
```
Comment (johnchen383) [inline package.json:13]: update to package.json should be done in client one as not all modules depend on this.

**Review by johnchen383 — CHANGES_REQUESTED** [review]
Thanks for those changes. Just one thing to fix up :))

**Review by johnchen383 — APPROVED** [review]
Lgtm

**Pull request merged**